modules/ossm-extensions-ref-wasmplugin.adoc

```diff
@@ -0,0 +1,176 @@
+////
+This module included in the following assemblies:
+*service_mesh_/v2x/ossm-extensions.adoc
+////
+:_content-type: REFERENCE
+[id="ossm-wasm-ref-wasmplugin_{context}"]
+= WasmPlugin API reference
+
+The WasmPlugins API provides a mechanism to extend the functionality provided by the Istio proxy through WebAssembly filters.
+
+You can deploy multiple WasmPlugins. The order of execution (as part of Envoy’s filter chain) is determined by the `phase` and `priority` settings, allowing the configuration of complex interactions between user-supplied WasmPlugins and Istio’s internal filters.
+
+In the following example, an authentication filter that implements an OpenID flow and populates the Authorization header with a JSON Web Token (JWT) to be consumed by Istio authentication is deployed to the ingress gateway. The WasmPlugin file lives in the proxy sidecar filesystem. Note the field `url`.
+
+[source,yaml]
+----
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: openid-connect
+  namespace: istio-ingress
+spec:
+  selector:
+    matchLabels:
+      istio: ingressgateway
+  url: file:///opt/filters/openid.wasm
+  sha256: 1ef0c9a92b0420cf25f7fe5d481b231464bc88f486ca3b9c83ed5cc21d2f6210
+  phase: AUTHN
+  pluginConfig:
+    openid_server: authn
+    openid_realm: ingress
+----
+
+In the following example, an authentication filter is deployed to the ingress-gateway that implements an OpenID flow and populates the `Authorization` header with an Oracle Cloud Infrastructure (OCI) image.
```

Comment:
Oracle is wrong here. But I'd suggest to rewrite this with something like:

Below is the same example, but this time an OCI image is used instead of a file in the filesystem. Note the fields `url`, `imagePullPolicy` and `imagePullSecret`.

Reply:
Bah, I looked up OCI and then wrote down the wrong definition for the acronym.   =(  It's Open Container Initiative.